build.gradle.kts

```
@@ -1,11 +1,50 @@
+import org.gradle.api.tasks.PathSensitivity.NAME_ONLY
+import org.jetbrains.kotlin.gradle.targets.js.nodejs.NodeJsRootPlugin.Companion.kotlinNodeJsRootExtension
+import org.jetbrains.kotlin.gradle.targets.js.yarn.yarn
+
 plugins {
     kotlin("multiplatform") version "2.1.10"
     kotlin("plugin.serialization") version "2.1.10"
     id("io.kotest.multiplatform") version "6.0.0.M1"
     distribution
 }
 
+val collectCatalogTypings by tasks.registering(Sync::class) {
+    from("../github-actions-typing-catalog/typings")
+    include("**/action-types.yml")
+    into(layout.buildDirectory.dir("catalog-typings"))
+    eachFile {
+        relativePath = RelativePath(
+            relativePath.isFile,
+            relativePath
+                .segments
+                .joinToString("_")
+                .replace("_action-types", "")
+        )
+    }
+    includeEmptyDirs = false
+}
+
+val collectExternalTypings by tasks.registering(Sync::class) {
+    mapOf(
+        "Vampire_setup-wsl_v1.yml" to "Vampire/setup-wsl/refs/heads/v1/action-types.yml",
+        "Vampire_setup-wsl_v2.yml" to "Vampire/setup-wsl/refs/heads/v1/action-types.yml",
+        "Vampire_setup-wsl_v3.yml" to "Vampire/setup-wsl/refs/heads/v1/action-types.yml",
+        "Vampire_setup-wsl_v4.yml" to "Vampire/setup-wsl/refs/heads/v1/action-types.yml",
```

```suggestion
        "Vampire_setup-wsl_v2.yml" to "Vampire/setup-wsl/refs/heads/v1/action-types.yml",
        "Vampire_setup-wsl_v3.yml" to "Vampire/setup-wsl/refs/heads/v2/action-types.yml",
        "Vampire_setup-wsl_v4.yml" to "Vampire/setup-wsl/refs/heads/v3/action-types.yml",
```

To avoid such issues, we can simply have a `listOf("v1", "v2", "v3", "v4")` and build the map out of them. I'm also fine with using just the newest major version here - should be good enough.

---

It was actually mainly to have a few examples and test-cases and was too lazy to lookup other actions providing typings. :-D I removed 1-3 and only kept v4 and master now.